docs/src/components/productX/XComponentsSwitcher.tsx

```diff
@@ -0,0 +1,173 @@
+import * as React from 'react';
+import dynamic from 'next/dynamic';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import KeyboardArrowRightRounded from '@mui/icons-material/KeyboardArrowRightRounded';
+import PivotTableChartRoundedIcon from '@mui/icons-material/PivotTableChartRounded';
+import CalendarMonthRoundedIcon from '@mui/icons-material/CalendarMonthRounded';
+import AccountTreeRounded from '@mui/icons-material/AccountTreeRounded';
+import BarChartRoundedIcon from '@mui/icons-material/BarChartRounded';
+import { visuallyHidden } from '@mui/utils';
+import Highlighter from 'docs/src/components/action/Highlighter';
+import Link from 'docs/src/modules/components/Link';
+import ROUTES from 'docs/src/route';
+
+const SwipeableViews = dynamic(() => import('react-swipeable-views'), { ssr: false });
+
+function ComponentItem({
+  label,
+  icon,
+  name,
+  description,
+  href,
+}: {
+  label: string;
+  icon: React.ReactNode;
+  name: React.ReactNode;
+  description?: React.ReactNode;
+  href: string;
+}) {
+  return (
+    <Box
+      component="span"
+      sx={{
+        display: 'flex',
+        p: 2,
+        flexDirection: { xs: 'column', md: 'row' },
+        alignItems: { md: 'center' },
+        gap: 2.5,
+      }}
+    >
+      <div>{icon}</div>
+      <div>
+        <Typography
+          component="span"
+          color="text.primary"
+          variant="body2"
+          fontWeight="bold"
+          display="block"
+        >
+          {name}
+        </Typography>
+        {description && (
+          <Typography
+            component="span"
+            color="text.secondary"
+            variant="body2"
+            fontWeight="regular"
+            display="block"
+            gutterBottom
+          >
+            {description}
+          </Typography>
+        )}
+        <Link
+          href={href}
+          color="primary"
+          variant="body2"
+          sx={{
+            display: 'inline-flex',
+            alignItems: 'center',
+            fontWeight: 'semiBold',
+            '& > svg': { transition: '0.2s' },
+            '&:hover > svg': { transform: 'translateX(2px)' },
+          }}
+          onClick={(event: React.MouseEvent<HTMLAnchorElement>) => {
+            event.stopPropagation();
+          }}
+        >
+          <span>View the docs</span>{' '}
+          <Box component="span" sx={visuallyHidden}>
+            {label}
+          </Box>
```

Comment:
I don't get the point of this label since it's hidden and it does not add extra information because the `label` text is exactly the same as the `description`

Reply:
@siriwatknp — any guidance here, Jun? I didn't remove it because I followed the same pattern as in the `ProductSwitcher` file, which has this invisible label element. Should we keep it or not?